### ISSUE TITLE
Bug: no-duplicate imports does not check for type of import ( default vs * import)

### ISSUE THREAD

**Issue by nopeless** [eslint/eslint#15441]
### Environment

Node version: v16.6.0
npm version: v7.19.1
Local ESLint version: v8.4.1
Global ESLint version: v8.1.0
Operating System: Windows 10

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

Set up two types of import

`import * as ... from "..."`

and

`import ... from "..."`

when exporting 
</details>

```js
// index.js
import lib, * as libProps from "./lib.js";

console.log(lib);
console.log(libProps);

// lib.js
import defaultExport from "./someimport.js";

export default 1;

export * as defaultExport from "./someimport.js";

// someimport.js
export default `I am default export`;

export const prop = `I am named export`;

```

### What did you expect to happen?

This should not trigger `no-duplicate-imports` since there is no way to import, store it in a variable, and export at the same time

```
// console output
I am default export
[Module: null prototype] {
  default: 'I am default export',
  defaultExport: [Module: null prototype] {
    default: 'I am default export',
    prop: 'I am named export'
  }
}
```

### What actually happened?

This is marked as an error (`no-duplicate-imports`)

`export * as defaultExport from "./someimport.js"`

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I'll probably ignore this rule if it doesn't get fixed

**Comment by mdjermanovic** [comment]
Hi @nopeless, thanks for the issue! 

Can you share your configuration for the `no-duplicate-imports` rule?

If you have `"includeExports": true` option, then this works as intended per [documentation for this option](https://eslint.org/docs/rules/no-duplicate-imports#options), because the example code can be rewritten to import from `./someimport.js` only once.

```js
import defaultExport, * as someImport from "./someimport.js";

export default 1;

export { someImport as defaultExport };
```

**Comment by mdjermanovic** [comment]
> // console output
> I am default export
> [Module: null prototype] {
>   default: 'I am default export',
>   defaultExport: [Module: null prototype] {
>     default: 'I am default export',
>     prop: 'I am named export'
>   }
> }

This doesn't look right for the given content of `lib.js`. It has `export default 1;`, so in `index.js` module `lib` and `libProps.default` should be `1`.

**Comment by nopeless** [comment]
> Hi @nopeless, thanks for the issue!
> 
> Can you share your configuration for the `no-duplicate-imports` rule?
> 
> If you have `"includeExports": true` option, then this works as intended per [documentation for this option](https://eslint.org/docs/rules/no-duplicate-imports#options), because the example code can be rewritten to import from `./someimport.js` only once.
> 
> ```js
> import defaultExport, * as someImport from "./someimport.js";
> 
> export default 1;
> 
> export { someImport as defaultExport };
> ```

Thanks, I think I missed `export { <import variable> as <name> }`

I only tried export <import variable> as <name>, which wouldn't work.

> This doesn't look right for the given content of lib.js. It has export default 1;, so in index.js module lib and libProps.default should be 1.

I changed `defaultExport` to `1` for a moment and forgot to reflect it in this issue

You are correct

Also, I didn't check all the options in the documentation. You are also correct here.
 I had `includeExports: true`

Thanks for explaining the issue 👍
